edxval/tests/test_api.py

```diff
@@ -1150,29 +1170,37 @@ def assert_encoded_video_matches_dict(self, encoded_video, encoded_video_dict):
     def assert_invalid_import(self, xml, course_id=None):
         edx_video_id = "test_edx_video_id"
         with self.assertRaises(ValCannotCreateError):
-            api.import_from_xml(xml, edx_video_id, course_id)
+            api.import_from_xml(xml, edx_video_id, self.file_system, constants.EXPORT_IMPORT_STATIC_DIR, course_id)
         self.assertFalse(Video.objects.filter(edx_video_id=edx_video_id).exists())
 
     def assert_transcripts(self, video_id, expected_transcripts):
         """
-        Compare `received` with `expected` and assert if not equal
+        Compare `received` with `expected` and assert if not equal.
         """
-        # Verify total number of expected transcripts for a video
+        # Verify total number of expected transcripts for a video.
         video_transcripts = VideoTranscript.objects.filter(video__edx_video_id=video_id)
         self.assertEqual(video_transcripts.count(), len(expected_transcripts))
 
-        # Verify data for each transcript
+        # Verify data for each transcript.
         for expected_transcript in expected_transcripts:
             language_code = expected_transcript['language_code']
-            expected_transcript['name'] = expected_transcript.pop('transcript')
 
-            # get the imported transcript and rename `url` key
-            received = api.TranscriptSerializer(
+            # Get the imported transcript and rename `url` key.
+            received_transcript = api.TranscriptSerializer(
                 VideoTranscript.objects.get(video__edx_video_id=video_id, language_code=language_code)
             ).data
-            received['name'] = received.pop('url')
 
-            self.assertDictEqual(received, expected_transcript)
+            # Assert transcript content
+            expected_transcript_content = expected_transcript.get('file_data', None)
+            received_transcript['file_data'] = json.loads(
+                api.get_video_transcript_data(video_id, language_code)['content']
+            )
+
+            # Remove not needed keys.
+            expected_transcript.pop('transcript', None)
```

Comment:
we can use: `assertDictContainsSubset` instead of popping items.

Reply:
actually `recieved_transcript.url` is saved with uuid so we can't check with `expected_transcript.url` which has static file url.